processing/src/main/java/io/druid/query/Query.java

```diff
@@ -61,8 +63,9 @@
   String SELECT = "select";
   String TOPN = "topN";
   String DATASOURCE_METADATA = "dataSourceMetadata";
+  String JOIN = "join";
 
-  DataSource getDataSource();
+  Iterable<DataSourceWithSegmentSpec> getDataSources();
```

Comment:
Any particular reason why getDataSources() returns Iterable instead of List or Collection? It would simplify code which currently uses StreamSupport.